### PR TITLE
fix(ci): canary test failure on npm install

### PR DESCRIPTION
**Pull request by pranavosu**
#### Description of changes

With the release of[ npm 11.0.0](https://github.com/npm/cli/releases/tag/v11.0.0), canary builds started failing with 

```Run npm install -g npm@latest
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: npm@11.0.0
npm ERR! notsup Not compatible with your version of node/npm: npm@11.0.0
npm ERR! notsup Required: {"node":"^20.17.0 || >=22.9.0"}
npm ERR! notsup Actual:   {"npm":"[10](https://github.com/aws-amplify/amplify-js/actions/runs/12356454590/job/34494840473#step:16:11).5.0","node":"v18.20.2"}
```

We don't need to install the latest version of npm, as we're setting up required node/npm versions earlier in the action. Removing `npm install -g npm@latest` fixes the issue.

#### Description of how you validated changes

- Run build script locally
- Run tests locally

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/callable-canary-sampleapp-tests.yml [hunk 1]
```diff
@@ -48,8 +48,8 @@ jobs:
         run: |
           cp credentials-auth.spec.js new-react-app.spec.js
         working-directory: amplify-js-samples-staging/cypress/integration/auth
-      - name: Install amplify
-        run: npm install amplify
+      - name: Install dependencies
+        run: npm install
```
Comment (AllanZhengYP) [inline .github/workflows/callable-canary-sampleapp-tests.yml:52]: Why is this needed?

Comment (pranavosu) [reply .github/workflows/callable-canary-sampleapp-tests.yml:52]: didn't mean to include this actually, just something else i found. i'll create a separate PR for this

**Review by AllanZhengYP — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]
